packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -28,6 +28,44 @@ const int _kDefaultSizeBytes = 100 << 20; // 100 MiB
 ///
 /// A shared instance of this cache is retained by [PaintingBinding] and can be
 /// obtained via the [imageCache] top-level property in the [painting] library.
+///
+/// {@tool sample}
+///
+/// This sample shows how to supply your own caching logic and replace the
+/// global [imageCache] varible.
+///
+/// ```dart
+/// import 'package:flutter/widgets.dart' show ImageCache, runApp, StatelessWidget,
```

Comment:
Since this is the only import now, I'd probably just import the entire file without the show to simplify this example further.

Reply:
Done